crates/sui-rpc-api/proto/sui/rpc/v2alpha/archival_service.proto

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package sui.rpc.v2alpha;
+
+import "google/protobuf/field_mask.proto";
+import "sui/rpc/v2alpha/checkpoint.proto";
+import "sui/rpc/v2alpha/executed_transaction.proto";
+import "sui/rpc/v2alpha/object.proto";
+
+// Alternitive names:
+// - HistoryService
+// - LedgerService
+// - HistoricalService
+// - HistoricalDataService
+service ArchivalService {
+  rpc GetObject(GetObjectRequest) returns (GetObjectResponse);
+  rpc BatchGetObjects(BatchGetObjectsRequest) returns (BatchGetObjectsResponse);
+
+  rpc GetTransaction(GetTransactionRequest) returns (GetTransactionResponse);
+  rpc BatchGetTransactions(BatchGetTransactionsRequest) returns (BatchGetTransactionsResponse);
+
+  rpc GetCheckpoint(GetCheckpointRequest) returns (GetCheckpointResponse);
+  rpc BatchGetCheckpoints(BatchGetCheckpointsRequest) returns (BatchGetCheckpointsResponse);
+}
+
+message GetObjectRequest {
+  // Required. The `ObjectId` of the requested object.
+  optional string object_id = 1;
+
+  // Request a specific version of the object.
+  // If no version is specified, and the object is live, then the latest
+  // version of the object is returned.
```

Comment:
Should we support the live version of this request here? Goes back to the question of whether this is generally easy to support over all stores that we would want to implement the historical service over.

Reply:
I don't really see why we shouldn't. In the same vein we probably should have a "bounded_version" field to request an object bounded by a version. 

The alternative would be having multiple rpcs for each of these and I think that sort of complicates usage of the api and leads to multiple "get object" rpcs.

Reply:
If we think that all the stores we want to implement this service over can support the necessary queries efficiently, then I don't see a problem with it, but I thought that wasn't the case.